pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
> "postCommit": {
>   "command": ["/any_command_that_acts as entrypoint"],
>   "args": ["args to entrypoint"]
> }

@mfojtik  this would translate to 

```json
{
  "postCommit": {
    "command": ["bash", "-c"],
    "args": ["rake test"]
  }
}
```

How do we expose that in the Web Console? It doesn't seem to translate well to a "type here" text area (as other CI tools have).

There's a known limitation about using Bash, but the limitation simplifies the user experience.

Our images have their [entrypoint](https://github.com/openshift/sti-base/blob/8d9514866b4f4600638d0a07603552b21aa77707/Dockerfile#L88) set to a bash script, which would work fine for people omitting the `command` field above.
However, for people using other images, we cannot force their images to have "friendly entrypoints", and we want to give a consistent experience across S2I and Docker builds independent of the image you are running.

We agreed on requiring / expecting that the images have `bash`, though. For the record, I argued whether this was a reasonable assumption.